src/Arcus.Messaging.Pumps.ServiceBus/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -197,6 +213,24 @@ public static IServiceCollection AddServiceBusTopicMessagePump<TMessagePump>(thi
             return services;
         }
 
+        /// <summary>
+        ///     Adds a message handler to consume messages from Azure Service Bus Topic
+        /// </summary>
+        /// <param name="topicName">Name of the topic to work with</param>
+        /// <param name="subscriptionName">Name of the subscription to process</param>
+        /// <param name="services">Collection of services to use in the application</param>
+        /// <param name="secretName">Name of the secret to retrieve using your registered <see cref="ISecretProvider"/> implementation</param>
+        /// <param name="configureMessagePump">Capability to configure how the message pump should behave</param>
+        /// <returns>Collection of services to use in the application</returns>
+        public static IServiceCollection AddServiceBusTopicMessagePump(this IServiceCollection services, string topicName, string subscriptionName, string secretName, Action<AzureServiceBusMessagePumpOptions> configureMessagePump = null)
+        {
+            Guard.NotNull(services, nameof(services));
+
+            AddServiceBusMessagePump<AzureServiceBusMessagePump>(services, topicName, subscriptionName, getConnectionStringFromSecretFunc: secretProvider => secretProvider.GetRawSecretAsync(secretName), configureMessagePump: configureMessagePump);
+
+            return services;
+        }
+
         /// <summary>
         ///     Adds a message handler to consume messages from Azure Service Bus Topic
```

Comment:
We should update all the `AddServiceBusTopicMessagePump` & `AddServiceBusQueueMessagePump` extensions:

```suggestion
        ///     Adds a message pump to consume messages from Azure Service Bus Topic
```

message handlers are what we add via WithMessagePumpHandler

Reply:
Yes